src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,373 @@
+            bool containsAwaitExpression,
+            TStatementSyntax statementContainsCallee,
+            IInvocationOperation invocationOperation,
+            bool removeCalleeDeclarationNode,
+            CancellationToken cancellationToken)
+        {
+            var methodParametersInfo = await MethodParametersInfo.GetMethodParametersInfoAsync(_syntaxFacts, document, invocationOperation, cancellationToken).ConfigureAwait(false);
+            var inlineContext = await GetInlineMethodContextAsync(
+                document,
+                calleeMethodInvocationNode,
+                calleeMethodSymbol,
+                inlineExpression,
+                statementContainsCallee,
+                methodParametersInfo,
+                cancellationToken).ConfigureAwait(false);
+            return await ChangeSolutionAsync(
+                document,
+                calleeMethodNode,
+                callerDeclarationNode,
+                callerSymbol,
+                containsAwaitExpression,
+                inlineContext,
+                removeCalleeDeclarationNode,
+                cancellationToken).ConfigureAwait(false);
+        }
+
+        private async Task<Solution> ChangeSolutionAsync(Document document,
+            TMethodDeclarationSyntax calleeMethodNode,
+            SyntaxNode callerDeclarationNode,
+            ISymbol callerSymbol,
+            bool containsAwaitExpression,
+            InlineMethodContext inlineMethodContext,
+            bool removeCalleeDeclarationNode,
+            CancellationToken cancellationToken)
+        {
+            var solution = document.Project.Solution;
+            var solutionEditor = new SolutionEditor(solution);
+            var documentEditor = await solutionEditor.GetDocumentEditorAsync(document.Id, cancellationToken).ConfigureAwait(false);
+            var statementContainsCalleeInvocationExpression = inlineMethodContext.StatementContainingCallee;
+            foreach (var statement in inlineMethodContext.StatementsToInsertBeforeCallee)
+            {
+                documentEditor.InsertBefore(
+                    statementContainsCalleeInvocationExpression,
+                    // Make sure the statement is aligned with the existing statement
+                    statement.WithTriviaFrom(statementContainsCalleeInvocationExpression));
+            }
+
+            var syntaxNodeToReplace = inlineMethodContext.SyntaxNodeToReplace;
+            var inlineSyntaxNode = inlineMethodContext.InlineSyntaxNode;
+            documentEditor.ReplaceNode(syntaxNodeToReplace, inlineSyntaxNode);
+
+            // If the inline content has 'await' expression, then make sure the caller is converted to 'async' method
+            if (containsAwaitExpression)
+            {
+                // If the inlineExpresion contains await Expression, check has been done to make sure
+                // the caller is Method
+                var callerMethodSymbol = (IMethodSymbol)callerSymbol;
```

Comment:
Why was `callerSymbol` not already typed to be an IMethodSymbol?

Reply:
It could be property